partiql-conformance-tests/build.rs

```diff
@@ -0,0 +1,296 @@
+use codegen::{Formatter, Function, Scope};
+use ion_rs::value::reader::{element_reader, ElementReader};
+use ion_rs::value::{Element, Sequence, Struct, SymbolToken};
+use ion_rs::IonType;
+use std::ffi::OsStr;
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use walkdir::WalkDir;
+
+const PASS_PARSE_TESTS_DIR: &str = "partiql-tests/partiql-test-data/pass/parser";
+const FAIL_PARSE_TESTS_DIR: &str = "partiql-tests/partiql-test-data/fail/parser";
+const PARSER_IMPORT: (&str, &str) = ("partiql_parser", "lalr_parse");
+
+fn main() {
+    let pass_parse_test_files: Vec<PathBuf> = all_ion_files_in(PASS_PARSE_TESTS_DIR);
+    let fail_parse_test_files: Vec<PathBuf> = all_ion_files_in(FAIL_PARSE_TESTS_DIR);
+
+    for pass_parse_test_file in &pass_parse_test_files {
+        let mut scope = create_pass_parse_scope(pass_parse_test_file);
+        scope.import(PARSER_IMPORT.0, PARSER_IMPORT.1);
+        let full_file_name = to_full_file_name("pass_parse", pass_parse_test_file);
+        let dest_path: PathBuf = ["tests", full_file_name.as_str()].iter().collect();
```

Comment:
Destination path is `tests/pass_parse_<file_name>.rs` (and is `tests/fail_parse_<file_name>.rs`. This is unconventional for build.rs scripts. From the [docs](https://doc.rust-lang.org/cargo/reference/build-script-examples.html):

>In general, build scripts should not modify any files outside of OUT_DIR. It may seem fine on the first blush, but it does cause problems when you use such crate as a dependency, because there's an implicit invariant that sources in .cargo/registry should be immutable. cargo won't allow such scripts when packaging.

Previously, I had generated the tests to a file in the `OUT_DIR`, which generates to the `target` folder. But there was limited IDE support to run the tests.

Reply:
I think this is ok for now. Essentially we're just harnessing the build script to generate the test code. 

The alternatives are either not be able to run the tests easily to debug or to use our own pre-processing step, which doesn't seem worth it.

Reply:
Just a food for thought: do we see using this package as a dependency in future?

Reply:
I don't think so. Especially if all of the parsing, model, & generation code are pulled out to a different crate.